e2e/app/component/card-base.ts

```diff
@@ -10,10 +10,26 @@ export default abstract class CardBase extends Container {
     super(page, xpath);
   }
 
+  /**
+   * @deprecated
+   * Replace with asElement()
+   */
   asElementHandle(): ElementHandle {
     return this.cardElement.asElement();
   }
 
+  async asElement(): Promise<ElementHandle | null> {
+    if (this.getXpath() !== undefined) {
+      // Refresh cardElement even if cardElement is already initialized.
```

Comment:
Can you add the "why?" to the comment here? Isn't it possible this is grabbing a different element than was intended by the test code?

Reply:
I honestly don't think it's possible same xpath will find a different element next time it is used in find because `this.page` and `this.getXpath()` don't change.
My main concern is test script may act on a stale element. An element can become stale, for example, when the  page or/and elements on page have reloaded automatically by javascript. Or the element has "moved" (thus changed) after new elements added or disappeared. If we attempt to click on a stale element, Puppeteer "remembers" the old element and it clicks on old element's position.

Reply:
If this is the case, why would we want to cache the element at all? Can we not just remove `this.cardElement` and access the value via `waitForXpath` everytime?